proxy/ParentConsistentHash.cc

```diff
@@ -58,29 +61,131 @@ ParentConsistentHash::~ParentConsistentHash()
   delete chash[SECONDARY];
 }
 
+void
+ParentConsistentHash::getPathHash_Helper(char *buffer, int size, const char *tmp, int len)
+{
+  int slen;
+  int max = size - 1;
+  slen    = (len > max) ? max : len;
+  strncpy(buffer, tmp, slen);
+  buffer[slen] = 0;
+}
```

Comment:
This should be a static function and not a class method IMO.

Also, I was somehow commenting on an old diff, but you should use the `MIN()` macro here too.